### PR TITLE
[Core] Moving flag from Condition/Element to GeometricalObject

### PR DESCRIPTION
**Pull request by loumalouomega**
Basically that. The flag is common between the geometries and conditions, but currently both derive from Flag instaed of bein included on the GeometricalObject

### CODE REVIEW

**Comment by loumalouomega** [comment]
As discussed on #4707

**Comment by pooyan-dadvand** [comment]
I agree with the change but let communicate it to the @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
similar to #3244

**Comment by loumalouomega** [comment]
> similar to #3244

Not exactly, but good point

File: kratos/includes/geometrical_object.h [hunk 1]
```diff
@@ -132,6 +140,25 @@ class GeometricalObject : public IndexedObject
         return *mpGeometry;
     }
 
+    ///@}
+    ///@name Flags
```
Comment (jcotela) [inline kratos/includes/geometrical_object.h]: This should go in the Access block, no need for a new one

File: kratos/includes/geometrical_object.h [hunk 2]
```diff
@@ -74,21 +75,27 @@ class GeometricalObject : public IndexedObject
     ///@{
 
     /// Default constructor.
-    explicit GeometricalObject(IndexType NewId = 0) : BaseType(NewId),
-        mpGeometry()
+    explicit GeometricalObject(IndexType NewId = 0)
+        : BaseType(NewId),
```
Comment (jcotela) [inline kratos/includes/geometrical_object.h]: I would suggest removing the `typedef IndexedObject BaseType` and replace `BaseType` with `IndexedObject` for clarity.

**Comment by jcotela** [comment]
I think this is a change for good, thanks for taking the time.

**Comment by loumalouomega** [comment]
> I think this is a change for good, thanks for taking the time.

Done, I also added the Doxygen doc and exposed properly the Flags of the geometrical object to python

**Review by pooyan-dadvand — APPROVED** [review]
@KratosMultiphysics/technical-committee approves it

**Pull request merged**